C++20 disallows template-ids (e.g. Foo<T>) in constructor and destructor
definitions inside a class template or explicit specialisation.  Remove the
redundant template argument from each ctor/dtor name.

--- a/src/vpux_compiler/src/dialect/IE/transforms/passes/adapt_shapes_for_scale_shift.cpp
+++ b/src/vpux_compiler/src/dialect/IE/transforms/passes/adapt_shapes_for_scale_shift.cpp
@@ -36,7 +36,7 @@ namespace {
 template <typename EltwiseOp>
 class BroadcastEltwiseRewriter final : public mlir::OpRewritePattern<EltwiseOp> {
 public:
-    BroadcastEltwiseRewriter<EltwiseOp>(mlir::MLIRContext* ctx, Logger log)
+    BroadcastEltwiseRewriter(mlir::MLIRContext* ctx, Logger log)
             : mlir::OpRewritePattern<EltwiseOp>(ctx), _log(log) {
         this->setDebugName("BroadcastEltwiseRewriter");
     }
@@ -205,7 +205,7 @@ namespace {
 template <typename EltwiseOp>
 class TransposeEltwiseRewriter final : public mlir::OpRewritePattern<EltwiseOp> {
 public:
-    TransposeEltwiseRewriter<EltwiseOp>(mlir::MLIRContext* ctx, Logger log)
+    TransposeEltwiseRewriter(mlir::MLIRContext* ctx, Logger log)
             : mlir::OpRewritePattern<EltwiseOp>(ctx), _log(log) {
         this->setDebugName("TransposeEltwiseRewriter");
     }
@@ -318,7 +318,7 @@ namespace {
 template <typename EltwiseOp>
 class MultiNonTrivialDimEltwiseRewriter final : public mlir::OpRewritePattern<EltwiseOp> {
 public:
-    MultiNonTrivialDimEltwiseRewriter<EltwiseOp>(mlir::MLIRContext* ctx, Logger log)
+    MultiNonTrivialDimEltwiseRewriter(mlir::MLIRContext* ctx, Logger log)
             : mlir::OpRewritePattern<EltwiseOp>(ctx), _log(log) {
         this->setDebugName("MultiNonTrivialDimEltwiseRewriter");
     }
--- a/src/vpux_compiler/src/dialect/IE/transforms/passes/adjust_fake_qdq_params.cpp
+++ b/src/vpux_compiler/src/dialect/IE/transforms/passes/adjust_fake_qdq_params.cpp
@@ -302,16 +302,16 @@ template <>
 class OpParamScaler<IE::FakeQuantizeOp> : public OpParamScalerBase {
     mlir::Operation* _updatedOp;

 public:
-    OpParamScaler<IE::FakeQuantizeOp>(): OpParamScalerBase(), _updatedOp(nullptr) {
+    OpParamScaler(): OpParamScalerBase(), _updatedOp(nullptr) {
     }
     virtual inline std::pair<mlir::LogicalResult, bool> operator()(mlir::Operation* op, mlir::PatternRewriter& rewriter,
                                                                    mlir::MLIRContext* ctx, TraversalDir traversalDir,
                                                                    float scale, OpParamdata::ScaleMode) override;
     virtual mlir::Operation* updatedOp() override {
         return _updatedOp;
     }
-    virtual ~OpParamScaler<IE::FakeQuantizeOp>() {
+    virtual ~OpParamScaler() {
     }
 };
--- a/src/vpux_compiler/src/dialect/IE/transforms/passes/convert_to_scale_shift.cpp
+++ b/src/vpux_compiler/src/dialect/IE/transforms/passes/convert_to_scale_shift.cpp
@@ -353,7 +353,7 @@ namespace {
 template <typename BiasTypeOp>
 class ConvertBiasToScaleShift final : public mlir::OpRewritePattern<BiasTypeOp> {
 public:
-    ConvertBiasToScaleShift<BiasTypeOp>(mlir::MLIRContext* ctx, mlir::PatternBenefit benefit, Logger log)
+    ConvertBiasToScaleShift(mlir::MLIRContext* ctx, mlir::PatternBenefit benefit, Logger log)
             : mlir::OpRewritePattern<BiasTypeOp>(ctx, benefit), _log(log) {
         this->setDebugName("ConvertBiasToScaleShift");
     }
--- a/src/vpux_compiler/include/vpux/compiler/conversion/passes/IE2VPU/convert_IE_to_VPU_NCE.hpp
+++ b/src/vpux_compiler/include/vpux/compiler/conversion/passes/IE2VPU/convert_IE_to_VPU_NCE.hpp
@@ -126,6 +126,6 @@ namespace {
 template <class ConcreteOp>
 class EltwiseToNCE final : public mlir::OpRewritePattern<ConcreteOp> {
 public:
-    EltwiseToNCE<ConcreteOp>(mlir::MLIRContext* ctx, VPU::EltwiseType opType, Logger log)
+    EltwiseToNCE(mlir::MLIRContext* ctx, VPU::EltwiseType opType, Logger log)
             : mlir::OpRewritePattern<ConcreteOp>(ctx), _opType(opType), _log(log) {
     }
--- a/src/vpux_compiler/src/NPU50XX/conversion/passes/convert_IE_to_VPU_NCE_strategy.cpp
+++ b/src/vpux_compiler/src/NPU50XX/conversion/passes/convert_IE_to_VPU_NCE_strategy.cpp
@@ -23,6 +23,6 @@ namespace vpux::arch50xx {
 template <typename ConcreteOp>
 class ReduceToNCE final : public mlir::OpRewritePattern<ConcreteOp> {
 public:
-    ReduceToNCE<ConcreteOp>(mlir::MLIRContext* ctx, VPU::ReduceType opType, config::ArchKind arch, Logger log)
+    ReduceToNCE(mlir::MLIRContext* ctx, VPU::ReduceType opType, config::ArchKind arch, Logger log)
             : mlir::OpRewritePattern<ConcreteOp>(ctx), _opType(opType), _arch(arch), _log(log) {
     }
